server/pbench/bin/index-pbench

```diff
@@ -287,20 +315,39 @@ class ResultData(object):
                 self.counters['not_valid_json_file'] += 1
                 continue
 
-            convert_to_float(results)
+            # There should always be a 'parameters' block.
+            # There is no reason for this to be an array, but it is: we
+            # just pick the 0^th (and only element) as the metadata for
+            # the document.
+            assert len(results['parameters']['benchmark'])
+            md = results['parameters']['benchmark'][0]
+            md.update({"iteration_name": iter_name,
+                       "iteration_number": iter_number,
+                       "sample": sample})
+            benchmark = md['benchmark_name']
+            # for each file, generate a sequence of documents and yield each one
+            for ts, src in ResultData.gen_sample_sources(results, benchmark):
+                # each generated result is a document, accompanied by a timestamp
+                # and some metadata.
+                source = _dict_const()
+                source['@metadata'] = self.run_metadata
+                source['@metadata'].update(md)
```

Comment:
I believe this `.update(md)` also updates `self.run_metadata`.  Do we want that?

Reply:
Absolutely not. I'll check and fix it.